pg_search/migrations/0001_enable.py

```diff
@@ -0,0 +1,26 @@
+from django.conf import settings
+from django.db import migrations
+
+
+class Migration(migrations.Migration):
+    dependencies = [
+        ("zerver", "0001_initial"),
+    ]
+
+    database_setting = settings.DATABASES["default"]
+
+    operations = [
+        migrations.RunSQL(
+            sql="""CALL paradedb.create_bm25(
+                    index_name => 'fts_pg_search_idx',
+                    schema_name => 'zulip',
+                    table_name => 'zerver_message',
+                    key_field => 'id',
+                    text_fields => paradedb.field('rendered_content',
+                                                  tokenizer => paradedb.tokenizer('en_stem'))
+                                   || paradedb.field('subject', tokenizer => paradedb.tokenizer('en_stem'))
```

Comment:
These fields can take an additional `fast => true` parameter here to be indexed as "fast fields" (https://docs.paradedb.com/api-reference/indexing/fast_fields), but I don't have a good understanding of whether this is right in our case or not. I suppose we can benchmark with/without `fast => true` on czo.

> A field that is indexed as fast is stored in a column-oriented fashion. Fast fields are necessary for [aggregations/faceting](https://docs.paradedb.com/api-reference/faceting/overview). They can also improve the query times of [filtering](https://docs.paradedb.com/api-reference/full-text/filtering) and BM25 scoring.

Reply:
Hmm. Our current design doesn't intend to use the scoring particularly, but I could imagine us extending the search UI to offer a "Sort by best match" option, and it's probably a pain to rebuild these indices.

The other thing I'm uncertain about is whether we want to continue squashing the subject and content into the same field, even though one is HTML and the other is plain-text, or if there's a better way to implement that detail.

Reply:
Fast fields store data in columnar format, rather than row format. This is ideal for fields which you may need to do a COUNT over, for example an `id` field.

Imagine your use case is an e-commerce catalog. You want to show the top 50 results, but tell the user there are X million products in the catalog. Doing a COUNT(id) would allow you to do this. Counts are very slow on row databases like Postgres, so we offer fast fields to store specific fields in columnar format.